sticker/src/tensorflow/lr.rs

```diff
@@ -94,22 +94,23 @@ impl ExponentialDecay {
 impl LearningRateSchedule for ExponentialDecay {
     fn compute_step_learning_rate(&mut self, global_step: usize) -> f32 {
         if global_step < self.warmup_steps {
-            (self.initial_lr / (self.warmup_steps as f32)) * global_step as f32
-        } else {
-            self.lr
+            return (self.initial_lr / (self.warmup_steps as f32)) * global_step as f32;
         }
-    }
 
-    fn compute_epoch_learning_rate(&mut self, epoch: usize, _last_score: f32) -> f32 {
+        // start decay after warmup
+        let decay_steps = global_step - self.warmup_steps;
```

Comment:
Isn't there an off-by one here? If `global_step == warmup_steps`, then there is a division by zero below?

Reply:
No, this is dividing 0 by `decay_steps`.

Reply:
Ah I see where the confusion is coming from. We have `self.decay_steps` and this local variable which have almost identical names. I'll change that.

Reply:
Ah, yes! It's much clearer now...